src/com/elixr/training/Main.java

```diff
@@ -0,0 +1,66 @@
+package com.elixr.training;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.StringTokenizer;
+import java.lang.String;
+
+public class Main {
+
+    public static void main(String[] args) {
+        if (args.length != 2) {
+            System.out.println("Argument missing"); //changed error message file path not valid to argument missing
+            return;
+        }
+        String inputfilepath = args[0]; //changed variable name from inputfile to inputfilepath
+        String searchword = args[1];  //changed variable name from wordsearch to searchword
+        File file = new File(inputfilepath);
+        System.out.println("Processing................");
+        String data = readFile(inputfilepath); //combined two lines
+        isSupportedFile(file);
+        searchword(data, searchword);
+
+    }
+
+    public static String readFile(String fileName) {
+        String data = null; //changes "" to null
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return data;
```

Comment:
If any exception occurred inside try, like IOException, then there is a possibility for data to become null. Should do a null check for data in the main method.

Reply:
I added the null check method